pyproject.toml

```diff
@@ -1,3 +1,183 @@
+# [build-system]
+# requires = ["scikit-build-core", "cython<=3.0.12"]
+# build-backend = "scikit_build_core.build"
+
+[build-system]
+requires = [
+    "hatchling",
+    "scikit-build-core",
+    "pybind11",
+    "cmake",
+    "ninja",
+    "cython<=3.0.12"
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "pynest-ng"
+version = "3.8.post0.dev5"
+description = "NEST is a simulator for spiking neural network models that focuses on the dynamics, size and structure of neural systems rather than on the exact morphology of individual neurons."
+keywords = ["nest", "simulator", "neuroscience", "neural", "neuron", "network", "ai", "spike", "spiking"]
+readme = "README.md"
+license.file = "LICENSE"
+authors = [
+  { name = "NEST Initiative", email = "users@nest-simulator.org" },
+]
+maintainers = [
+  { name = "Steffen Graber", email = "s.graber@fz-juelich.de" },
+]
+requires-python = ">=3.9, <4.0"
+classifiers = [
+  "Development Status :: 3 - Alpha",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
+  "Operating System :: POSIX :: Linux",
+  "Operating System :: MacOS",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Programming Language :: C++",
+  "Topic :: Scientific/Engineering",
+  "Topic :: Scientific/Engineering :: Medical Science Apps.",
+  "Topic :: Scientific/Engineering :: Artificial Intelligence",
+  "Typing :: Typed",
+]
+dependencies = [
+  "numpy",
+  "pynest-api",
+  "pandas",
+  "scipy",
+  "matplotlib",
+  "h5py",
+  "setuptools",
+]
+
+[tool.hatch.build.targets.wheel.hooks.scikit-build]
+experimental = true
+
+# [tool.hatch.build.hooks.custom]
+# path = "build_support/pip_post-install.py"
+
+[project.optional-dependencies]
+nestml = [
+  "nestml"
+]
+mpi = [
+  "mpi4py"
+]
+
+[project.urls]
+Homepage = "https://nest-simulator.org/"
+Documentation = "https://nest-simulator.readthedocs.io/en/stable/"
+"Bug Tracker" = "https://github.com/nest/nest-simulator/issues"
+Changelog = "https://nest-simulator.readthedocs.io/en/stable/whats_new/index.html"
+
+[tool.scikit-build]

```

Comment:
Why "scikit"?

Reply:
I use a program called "Scikit-build-core" to help me package pynest-ng. "It provides a bridge between CMake and the Python build system, allowing you to make Python modules with CMake." https://scikit-build-core.readthedocs.io/en/latest/
The scikit tag here marks some definitions for the build process.